client/ayon_core/hosts/zbrush/api/pipeline.py

```diff
@@ -203,10 +211,24 @@ def containerise(
 
 
 def save_current_workfile_context(context):
+    """Function to save current workfile context data
+
+    Args:
+        context (dict): context data
+
+    """
     return write_context_metadata(ZBRUSH_SECTION_NAME_CONTEXT, context)
 
 
 def write_context_metadata(metadata_key, context):
+    """Write context data into the related json
+    which stores in zbrush_metadata folder
+    in the project work directory.
```

Comment:
As stated on Discord - the biggest issue I had with these functions was that some seemed to write into:

```
.zbrush_metadata/{workfile}/...
```

Whereas others seemed to read/write directly to
```
.zbrush_metadata/...
```

It seemed like the first was intended to be:
- Used when saving/opening the workfile (so like persisting and reading files)

And the second was intended to be:
- Store the 'in-memory' context, instances and other metadata to the current zbrush process. 
- Then on workfile save this is the data that would get copied/saved into the `.zbrush_metadata/{workfile}/...` folder
- Then on workfile open this is the data that would be replaced by any `.zbrush_metadata/{workfile}/...` of the workfile being opened.

If there is indeed this difference between writing to either of these two locations - then it's best to clarify in the docstring what the intended use case is of the function. The fact that `save_current_workfile_context` "saves current workfile context data" and that it is a function for it - tells me nothing more from the docstring than then function name. The docstring should instead be describing what it does and why.

E.g.
```python
def save_current_workfile_context(context):
    """Save current workfile context data to `.zbrush_metadata/{workfile}/key`

    This persists the current in-memory context to be set for a specific
    workfile on disk. Usually used on save to persist the local sessions'
    workfile context on save.

    The context data includes things like the project name, asset name,
    etc.

    Args:
        context (dict): context data
    """
```
This description might be inaccurate - I'm just writing out what I _think_ may be the case. In a way I hope I'm wrong in my description because that would actually showcase how limitied information I get from the function alone.



I see a `write_workfile_metadata` and `write_context_metadata` which I feel are the differences between writing to one of the two folders. They should really explain why they differ, what their intent is, etc.

Reply:
Just to clarify:

- I feel like all data should be related to a workfile, there's nothing else than persisting data with the workfile.
- The only other data is the data in-memory for the current session - that is either data that would not need to persist (since it's not related to a workfile) or is just 'in-memory' up until it gets persisted to disk on workfile save.

How the functions used here differ between these two - and whether that's the case - I have no idea.

Reply:
I agree with you that we need to further elaborate the docstring and we should rename the function for better scope on what the function does in the integration. 
I guess we can also mention why the zbrush_metadata folder is there and how it works in the integration (in admin docs) in the documentatio.